### PR TITLE
feat(RolesTable): add bulk Delete Roles button

### PR DESCRIPTION
**Pull request by InsaneZein**
### Description
Finishes up [RHCLOUD-32241](https://issues.redhat.com/browse/RHCLOUD-32241) by allowing the user to delete multiple roles at once if multiple rows are selected, and also fixes the bulk select issue in the Roles table. 

---

### Screenshots
Screenshot from 2024-11-19 14-33-14

---

### Checklist ☑️
- [ ] PR only fixes one issue or story 
- [ ] Change reviewed for extraneous code 
- [ ] UI best practices adhered to 
- [ ] Commits squashed and meaningfully named 
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

**Comment by InsaneZein** [comment]
> 9 snapshots failed from 4 test suites. Inspect your code changes or run `npm test -- -u` to update them.

I've updated the snapshots already, but it still says this. All the tests pass locally.

File: src/smart-components/role/RolesTable.tsx [hunk 1]
```diff
@@ -174,6 +185,18 @@ const RolesTable: React.FunctionComponent<RolesTableProps> = ({ selectedRole })
...
             }
+            actions={
+              <ResponsiveActions breakpoint="lg" ouiaId={`${ouiaId}-actions-dropdown`}>
+                <ResponsiveAction
+                  isDisabled={selected.length === 0 || selected.some(isRowSystemOrPlatformDefault)}
+                  onClick={() => {
+                    handleModalToggle(roles.filter((role) => selected.some((selectedRow: DataViewTrObject) => selectedRow.id === role.uuid)));
+                  }}
+                >
+                  Delete Roles
```
Comment (fhlavac) [inline src/smart-components/role/RolesTable.tsx:196]: This shoul dbe localized

**Comment by InsaneZein** [comment]
@fhlavac that worked. Thanks!

**Review by fhlavac — APPROVED** [review]

**Pull request merged**